asreview/webapp/api/auth.py

```diff
@@ -0,0 +1,512 @@
+
+
+@bp.route("/get_profile", methods=["GET"])
+@asreview_login_required
+def get_profile():
+
+    user = User.query.get(current_user.id)
+    if user:
+        result = (
+            200,
+            {
+                "email": user.email,
+                "origin": user.origin,
+                "name": user.name,
+                "affiliation": user.affiliation,
+                "public": user.public,
+            },
+        )
+    else:
+        result = (404, "No user found")
+
+    status, message = result
+    response = jsonify({"message": message})
+    return response, status
+
+
+@bp.route("/forgot_password", methods=["POST"])
+def forgot_password():
+
+    if current_app.config.get("EMAIL_VERIFICATION", False):
+
+        # get email address from request
+        email_address = request.form.get("email", "").strip()
+
+        # check if email already exists
+        user = User.query.filter(
+            or_(User.identifier == email_address, User.email == email_address)
+        ).one_or_none()
+
+        if not user:
+            result = (404, f'User with email "{email_address}" not found.')
+        elif user.origin != "asreview":
+            result = (404, f"Your account has been created with {user.origin}")
+        else:
+            # set a token
+            user = user.set_token_data(
+                current_app.config["SECRET_KEY"],
+                current_app.config["SECURITY_PASSWORD_SALT"],
+            )
+            try:
+                # store data
+                DB.session.commit()
+                # send email
+                send_forgot_password_email(user, current_app)
+                # result
+                result = (200, f"An email has been sent to {email_address}")
+
+            except SQLAlchemyError as e:
+                DB.session.rollback()
+                result = (403, f"Unable to to confirm user! Reason: {str(e)}")
+
+    status, message = result
+    response = jsonify({"message": message})
+    return response, status
+
+
+@bp.route("/update_profile", methods=["POST"])
+@asreview_login_required
+def update_profile():
+    """Update user profile"""
+    user = User.query.get(current_user.id)
+    if user:
+        email = request.form.get("email", "").strip()
+        name = request.form.get("name", "").strip()
+        affiliation = request.form.get("affiliation", "").strip()
+        password = request.form.get("password", None)
+        public = bool(int(request.form.get("public", "1")))
+
+        try:
+            user = user.update_profile(email, name, affiliation, password, public)
+            DB.session.commit()
+            result = (200, "User profile updated")
+        except IntegrityError as e:
+            DB.session.rollback()
+            result = (500, f"Unable to update your profile! Reason: {str(e)}")
+        except SQLAlchemyError as e:
+            DB.session.rollback()
+            result = (500, f"Unable to update your profile! Reason: {str(e)}")
```

Comment:
Sometimes an SQLAlchemyError gives status_code 500, sometimes 403. We should probably make these and the corresponding messages a bit more uniform.